src/main/scala/WeaviateOptions.scala

```diff
@@ -22,13 +22,29 @@ class WeaviateOptions(config: CaseInsensitiveStringMap) extends Serializable {
   val retries: Int = config.getInt(WEAVIATE_RETRIES_CONF, DEFAULT_RETRIES)
   val retriesBackoff: Int = config.getInt(WEAVIATE_RETRIES_BACKOFF_CONF, DEFAULT_RETRIES_BACKOFF)
   val timeout: Int = config.getInt(WEAVIATE_TIMEOUT, DEFAULT_TIMEOUT_SECONDS)
+  val oidcUsername: String = config.getOrDefault(WEAVIATE_OIDC_USERNAME, "")
+  val oidcPassword: String = config.getOrDefault(WEAVIATE_OIDC_PASSWORD, "")
+  val oidcClientSecret: String = config.getOrDefault(WEAVIATE_OIDC_CLIENT_SECRET, "")
+  val oidcAccessToken: String = config.getOrDefault(WEAVIATE_OIDC_ACCESS_TOKEN, "")
+  val oidcAccessTokenLifetime: Long = config.getLong(WEAVIATE_OIDC_ACCESS_TOKEN_LIFETIME, 0)
+  val oidcRefreshToken: String = config.getOrDefault(WEAVIATE_OIDC_REFRESH_TOKEN, "")
 
   var client: WeaviateClient = _
 
   def getClient(): WeaviateClient = {
     if (client != null) return client
     val config = new Config(scheme, host, null, timeout, timeout, timeout)
-    client = new WeaviateClient(config)
+
+    if (!oidcUsername.isBlank() && !oidcPassword.isBlank()) {
+      client = WeaviateAuthClient.clientPassword(config, oidcUsername, oidcPassword, null)
```

Comment:
What about scopes? Do people sometimes need to set those?

Reply:
They might sometimes need to pass custom `scopes` but that's a rare case.
I will add a ticket for adding a support for `scopes` and also `headers` bc for headers we are also passing `null` but sometimes people might need to pass some custom headers like for example `X-OpenAI-Api-Key:THE-KEY`

Reply:
I have added #73 issue to do this in a separate PR.